### PR TITLE
Add Service Catalog operator types

### PR DESCRIPTION
**Pull request by jboyd01**
cc @derekwaynecarr

### CODE REVIEW

File: operator/v1/types_openshiftservicecatalogapiserver.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+type OpenShiftServiceCatalogAPIServerSpec struct {
+	OperatorSpec `json:",inline"`
+
+	//enabled specifies if Service Catalog should be installed and started in the cluster
+	Enabled bool `json:"enabled"`
+
+	//serverArguments are key value pairs that will override the default set of arguments.  ServerArguments may include Feature Gates & Admission Plugins.  If you specify an invalid argument the server will not start. These values may cause invalid configurations.",
+	ServerArguments map[string][]string `json:"serverArguments"`
```
Comment (derekwaynecarr) [inline operator/v1/types_openshiftservicecatalogapiserver.go]: Why do we need to expose this?

File: operator/v1/types_openshiftservicecatalogcontrollermanager.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+type OpenShiftServiceCatalogControllerManagerSpec struct {
+	OperatorSpec `json:",inline"`
+
+	//enabled specifies if Service Catalog should be installed and started in the cluster
+	Enabled bool `json:"enabled"`
+
+	//serverArguments are key value pairs that will override the default set of arguments.  ServerArguments may include Feature Gates & Admission Plugins.  If you specify an invalid argument the server will not start. These values may cause invalid configurations.",
+	ServerArguments map[string][]string `json:"serverArguments"`
```
Comment (derekwaynecarr) [inline operator/v1/types_openshiftservicecatalogcontrollermanager.go]: Why do we need to expose this?

File: operator/v1/types_openshiftservicecatalogapiserver.go (hunk 1 above)
Comment (jboyd01) [reply operator/v1/types_openshiftservicecatalogapiserver.go]: initially added to enable cluster admins to configure with different sets of feature flags and provide flexibility.  Discussed with Derek and agree it provides to much flexibility, if needed specific feature gates should be exposed.  Alternatively, admins could go on their own and set to non managed/unsupportedConfigOverride

**Comment by derekwaynecarr** [comment]
recording discussion:

- drop the OpenShift prefix
- eliminate enabled in favor of management state

**Comment by derekwaynecarr** [comment]
thanks for the updates.

/approve
/lgtm

**Pull request merged**